microsoft-edge/devtools-guide-chromium/whats-new/2021/04/devtools.md

```diff
@@ -166,17 +166,215 @@ The [Microsoft Edge Tools for Visual Studio Code extension][VisualstudioMarketpl
+1.  Navigate to the **Scope** section.  
+    
+To review the history of this feature in the Chromium open-source project, navigate to Issue [11374][CR11374].  
+
+<!--  
+:::image type="complex" source="../../media/2021/04/dPDCek3EhZgLQPGtEG3y0fTn4v82/RCw19qd1bZOeNFydsUCW.msft.png" alt-text="JavaScript private brand checks" lightbox="../../media/2021/04/dPDCek3EhZgLQPGtEG3y0fTn4v82/RCw19qd1bZOeNFydsUCW.msft.png":::
+   JavaScript private brand checks  
+:::image-end:::  
+-->  
+
+#### Enhanced support for breakpoints debugging  
+
+In Microsoft Edge version 90 or earlier, DevTools only set breakpoints on one raw location.  In Microsoft Edge version 91 or later, DevTools properly set breakpoints in multiple scripts.  This feature is a modern support code splitting feature.  Multiple routes may include one shared component and are named code splits.  JavaScript bundlers include [Webpack][WebpackJsMain], and [Rollup][RollupjsMain].  In Microsoft Edge version 91 or later, DevTools sets breakpoints in all relevant locations correctly.  To review the history of this feature in the Chromium open-source project, navigate to Issues [1142705][CR1142705], [979000][CR979000], and [1180794][CR1180794].  
+
+#### Support hover preview with [] notation  
+
+DevTools now support hover preview on JavaScript member expressions that use the `[]` notation in the **Sources** tool.  To review the history of this feature in the Chromium open-source project, navigate to Issue [1178305][CR1178305].  
+
+<!--  
+:::image type="complex" source="../../media/2021/04/dPDCek3EhZgLQPGtEG3y0fTn4v82/XFxlQjvLVWJUiV24zDw2.msft.png" alt-text="Support hover preview with [] notation" lightbox="../../media/2021/04/dPDCek3EhZgLQPGtEG3y0fTn4v82/XFxlQjvLVWJUiV24zDw2.msft.png":::
+   Support hover preview with `[]` notation  
+:::image-end:::  
+-->  
+
+#### Improved outline of HTML files  
+
+DevTools now has better outline support for `.html` files.  In the **Sources** tool, open the `.html` file.  To turn on \(or off\) the code outline, select `Ctrl`+`Shift`+`O` on Windows/Linux or `Cmd`+`Shift`+`O` on macOS.  In the following figure, DevTools now correctly list all functions in the outline.  Previously, DevTools only displayed some of the functions.  To review the history of this feature in the Chromium open-source project, navigate to Issues [761019][CR761019] and [1191465][CR1191465].  
+
+:::image type="complex" source="../../media/2021/04/sources-pages-html-code-outline.msft.png" alt-text=" Improved outline of HTML files" lightbox="../../media/2021/04/sources-pages-html-code-outline.msft.png":::
+   Improved outline of HTML files  
+:::image-end:::  
+
+#### Proper error stack traces for Wasm debugging  
+
+In Microsoft Edge version 90 or earlier, DevTools only displayed generic Wasm references in the Error stack traces.  
+In Microsoft Edge version 91 or later, DevTools resolves inline function requests and displays proper error stack traces for Wasm debugging.  
+
+:::row:::
```

Comment:
For this image, navigate to: https://emscripten-dbg-stories.netlify.app/inlining-dwarf.html. Open DevTools > **Sources**. Select `Ctrl`+`P` and enter `inlining-dwarf.wasm`. Set a breakpoint at `0x0028a`. Refresh the page. When the breakpoint is hit, switch to the **Console** and then enter `new Error` and run that. The error now contains the correct stack trace. Here's what I got: 


![image](https://user-images.githubusercontent.com/34316555/117219123-75181680-adb9-11eb-9031-a356f1f50df3.png)

Reply:
Updated.